gradle-plugin/src/main/kotlin/dev/shreyaspatil/composeCompilerMetricsGenerator/plugin/ReportGenPlugin.kt

```diff
@@ -41,16 +46,99 @@ class ReportGenPlugin : Plugin<Project> {
-            }.getOrElse { error("This plugin is only applicable for Android modules") }
+            }.getOrNull()
+        val jvm =
+            runCatching {
+                target.extensions.getByType(KotlinJvmProjectExtension::class.java)
+            }.getOrNull()
+        val multiplatform =
+            runCatching {
+                target.extensions.getByType(KotlinMultiplatformExtension::class.java)
+            }.getOrNull()
+        val composeMultiplatform =
+            runCatching {
+                target.extensions.getByType(ComposeExtension::class.java)
+            }.getOrNull()
+
+        when {
+            composeMultiplatform != null -> {
+                when {
+                    jvm != null -> { // if kotlin jvm is applied
+                        target.configureKotlinJvmComposeCompilerReports(jvm)
+                    }
+                    multiplatform != null -> { // if kotlin multiplatform is applied
+                        target.configureKotlinMultiplatformComposeCompilerReports(multiplatform)
+                    }
+                    android != null -> { // if kotlin android is applied
+                        target.configureKotlinAndroidComposeCompilerReports(android)
+                    }
+                }
+            }
+            android != null -> {
+                target.configureKotlinAndroidComposeCompilerReports(android)
+            }
+        }
+    }
+
+    private fun Project.configureKotlinJvmComposeCompilerReports(jvmExt: KotlinJvmProjectExtension) {
+        // Create gradle tasks for generating report
+        registerComposeCompilerReportGenTaskForJvmProject(jvmExt.target.project.name)
+
+        afterEvaluate {
+            // When this method returns true it means gradle task for generating report is executing otherwise
+            // normal compilation task is executing.
+            val isFromReportGenGradleTask = executingComposeCompilerReportGenerationGradleTask()
+            if (isFromReportGenGradleTask) {
+                jvmExt.target {
+                    compilations.filter { !it.name.endsWith("Test") }.forEach {
+                        it.kotlinOptions {
+                            configureKotlinOptionsForComposeCompilerReport(this@configureKotlinJvmComposeCompilerReports)
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private fun Project.configureKotlinMultiplatformComposeCompilerReports(multiplatformExt: KotlinMultiplatformExtension) {
+        // Create gradle tasks for generating report
+        multiplatformExt.targets.configureEach {
+            if (this.name == "metadata") return@configureEach
+
+            if (this.name == "android") {
+                // register a task for each build type
+                registerComposeCompilerReportGenTaskForTarget(this, buildType = "Debug")
+                registerComposeCompilerReportGenTaskForTarget(this, buildType = "Release")
```

Comment:
So we don't get all variants automatically just like we get for android plugins? I mean, there should be no need for hardcoding "debug", "release" is what I think.

Reply:
This is a good question. The point is that Kotlin multiplatform plugin doesn't support build flavors (yet) so I thought it should be fine to hardcode the "Debug" and "Release" variants, but I think we can have two better solutions here:

1. Register a single task for android target in a multiplatform project just like we do it for any other target, the rationale behind this is in a multiplatform project it is uncommon to have different composable functions between build types. (at least this is what I've seen so far.)
2. make the configuration function accept an optional android extension parameter, and if the current target is android we can query the variants from android extension using `onVariants` and register tasks the same way we would do it for pure android modules.

What do you think? For me, I think I'll go with the second approach

Reply:
Implemented the second approach & it worked well.

Reply:
great!